cmd/translatesfx/translatesfx/otel.go

```diff
@@ -524,21 +524,59 @@ func saMonitorToRCReceiver(monitor map[string]interface{}, observers []interface
 		warnings = append(warnings, err)
 	}
 	delete(monitor, discoveryRule)
-	return map[string]map[string]interface{}{
-		key: {
+
+	cmp = component{
+		provisionalKey: key,
+		attrs: map[string]interface{}{
 			"rule":   rcr,
 			"config": monitor,
 		},
-	}, warnings
+	}
+	return
+}
+
+type component struct {
+	attrs          map[string]interface{}
+	provisionalKey string
```

Comment:
nit - how is this provisional? isn't it more of a ~`baseType` or `baseName`?

Reply:
Sure -- changed to `baseName` and added a comment.